src/app/manager/annual-leave/annual-leaves/annual-leave.component.ts

```diff
@@ -0,0 +1,44 @@
+import { IAnnualLeave } from './../model/annual-leave.model';
+import { Router } from '@angular/router';
+import { AnnualLeaveService } from '../services/annual-leave.service';
+import { Component, OnInit } from '@angular/core';
+import { MatTableDataSource } from '@angular/material/table';
+
+@Component({
+  selector: 'app-annual-leave',
+  templateUrl: './annual-leave.component.html',
+  styleUrls: ['./annual-leave.component.css']
+})
+export class AnnualLeaveComponent implements OnInit {
+
+  public dataSource = new MatTableDataSource<IAnnualLeave>();
+  public displayedColumns = ['Doctor', 'Starting', 'Ending', 'State', 'Urgent', 'Review' ];
+  public annualLeaves: IAnnualLeave[] = [];
+  constructor(private annualLeaveService:AnnualLeaveService, private router:Router) { }
+
+  ngOnInit(): void {
+    this.annualLeaveService.getAnnualLeaves().subscribe(res => {
+      this.annualLeaves = res;
+      console.log(res);
+      for(let i = 0; i < this.annualLeaves.length; i++){
```

Comment:
You can refactor this to use typescript map function. this.leaves = leaves.map(leave => {...})